extension-manifest-v3/src/background/stats.js

```diff
@@ -186,12 +186,40 @@ export function updateTabStats(tabId, requests) {
       );
     }
 
-    // After navigation stats are cleared, so the current `stats` variable might be outdated
-    if (requestsUpdated && stats === tabStats.get(tabId)) {
-      tabStats.set(tabId, stats);
+    if (requestsUpdated) {
+      if (
+        __PLATFORM__ === 'safari' &&
+        chrome.declarativeNetRequest.getMatchedRules
+      ) {
+        try {
+          const { rulesMatchedInfo } =
+            await chrome.declarativeNetRequest.getMatchedRules({
+              tabId,
+              minTimeStamp: stats.timestamp,
+            });
+
+          for (const info of rulesMatchedInfo) {
+            for (const tracker of stats.trackers) {
+              for (const request of tracker.requests) {
+                if (request.url === info.request.url) {
```

Comment:
this is great

please make it even better and add update stats for all urls that haven't matched already observed urls